src/lib/protocols/dns.c

```diff
@@ -368,6 +368,7 @@ static void ndpi_search_dns(struct ndpi_detection_module_struct *ndpi_struct, st
     int invalid = search_valid_dns(ndpi_struct, flow, &dns_header, payload_offset, &is_query);
     ndpi_protocol ret;
     u_int num_queries, idx;
+    char _hostname[256];
```

Comment:
Why not `sizeof(host_server_name)` ?

Reply:
I think you mean `sizeof(flow->host_server_name)`... Anyway the basic idea is to always extract the full hostname in the dns code and than let `ndpi_hostname_sni_set()` to save the *last* 80 bytes (if longer). This way, domain matching shouldn't fail even with long hostnames.
If we use `_hostname[80]` we extract the *first* 80 bytes of the hostname.
We can obviously use `_hostname[80]` *and* modify DNS dissector to extract the *last* 80 characters itself but it seems unnecessarily complicated, IMO

Reply:
Totally overlooked the string matching automata. Thanks for he hint.